parallel/client.py

```diff
@@ -89,6 +100,12 @@ def submit(self, task_fn, *args):
         self._idx = (self._idx + 1) % self.max_workers
         return future
 
+    def verify(self):
+        """
+        See abstract class for documentation.
+        """
+        return
```

Comment:
Better to raise `NotImplementedError` here instead.

Reply:
But doesn't that make it that we can't treat the non-abstract classes as the same? As I see it there is no validation to do here, it is 'good'.

Reply:
My thought is you can do the following

```
if client_type == "process":
   client = ProcessPoolClient()
elif client_type == "cuda":
   client = CudaClient()
 elif client_type == "grpc":
   client = GrpcClient()
 client.verify()
 ```

Reply:
Sorry you're totally right this was a brainfart on my end - for some reason I thought the ProcessPoolClient was the abstract class! (I hardly ever use it)